web/src/features/automations/server/router.ts

```diff
@@ -0,0 +1,479 @@
+import { createTRPCRouter } from "@/src/server/api/trpc";
+import { protectedProjectProcedure } from "@/src/server/api/trpc";
+import { z } from "zod/v4";
+import {
+  ActionCreateSchema,
+  ActionType,
+  JobConfigState,
+} from "@langfuse/shared";
+import { throwIfNoProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
+import { v4 } from "uuid";
+import {
+  getActionById,
+  getAutomations,
+  getAutomationById,
+  getConsecutiveAutomationFailures,
+  logger,
+} from "@langfuse/shared/src/server";
+import { generateWebhookSecret } from "@langfuse/shared/encryption";
+import { processWebhookActionConfig } from "./webhookHelpers";
+import { TRPCError } from "@trpc/server";
+import { auditLog } from "@/src/features/audit-logs/auditLog";
+
+export const CreateAutomationInputSchema = z.object({
+  projectId: z.string(),
+  name: z.string().min(1, "Name is required"),
+  eventSource: z.string(),
+  eventAction: z.array(z.string()),
+  filter: z.array(z.any()).nullable(),
+  status: z.enum(JobConfigState).default(JobConfigState.ACTIVE),
+  // Action fields
+  actionType: z.enum(ActionType),
+  actionConfig: ActionCreateSchema,
+});
+
+export const UpdateAutomationInputSchema = CreateAutomationInputSchema.extend({
+  automationId: z.string(),
+});
+
+export const automationsRouter = createTRPCRouter({
+  // Get automations that were recently auto-disabled due to failures
+  getCountOfConsecutiveFailures: protectedProjectProcedure
+    .input(
+      z.object({
+        projectId: z.string(),
+        automationId: z.string(),
+      }),
+    )
+    .query(async ({ ctx, input }) => {
+      // Check if user has at least read access to automations
+      throwIfNoProjectAccess({
+        session: ctx.session,
+        projectId: input.projectId,
+        scope: "automations:read",
+      });
```

Comment:
Shall we make this a property of the protectedProjectProcedure? Maybe as a future refactoring? IMO handling scope checks there could make everything simpler.